src/app/code/community/Shopgate/Cloudapi/Block/AnalyticsLogPurchase.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Block_AnalyticsLogPurchase extends Mage_Adminhtml_Block_Template
+{
+    /**
+     * @var int
+     */
+    protected $orderId;
+
+    /**
+     * @param int $orderId
+     */
+    public function setOrderId($orderId)
+    {
+        $this->orderId = $orderId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getJsonOrderData()
+    {
+        /** @var Mage_Sales_Model_Order $order */
+        $order  = Mage::getModel('sales/order')->load($this->orderId);
+        $result = new Varien_Object();
+        $result->setType(Shopgate_Cloudapi_Model_Order_Source::SOURCE_WEBCHECKOUT);
```

Comment:
Create a class for this. You will need to use ```\@method``` declarations instead of implementing getters/setters. You know, same as we had done in class ```Shopgate_Cloudapi_Model_Auth_Code```. Will also need to create an item class as well.